File: src/bots/random.js
// Random bot — picks a uniformly random valid column. Used as baseline.
import { validCols } from '../game/rules.js';

/**
 * @param {number[][]} board
 * @returns {number} column index
 */
export function randomBot(board) {
  const cols = validCols(board);
  return cols[Math.floor(Math.random() * cols.length)];
}

File: src/bots/random.py
# Random bot — picks a uniformly random valid column. Used as baseline.
import random as _random
from ..game.rules import valid_cols


def random_bot(board, token=None):
    """
    Pick a uniformly random valid column.
    token is accepted for interface compatibility but ignored.
    """
    cols = valid_cols(board)
    return _random.choice(cols)
